lib/datadog/ci/ext/environment.rb

```diff
@@ -312,6 +319,106 @@ def extract_bitrise(env)
+          }
+        rescue => e
+          Datadog.logger.debug("Unable to read git commit users: #{e.message} at #{e.backtrace.first}")
+          nil
+        end
+
+        def git_repository_url
+          exec_git_command('git ls-remote --get-url')
+        rescue => e
+          Datadog.logger.debug("Unable to read git repository url: #{e.message} at #{e.backtrace.first}")
+          nil
+        end
+
+        def git_commit_message
+          exec_git_command('git show -s --format=%s')
+        rescue => e
+          Datadog.logger.debug("Unable to read git commit message: #{e.message} at #{e.backtrace.first}")
+          nil
+        end
+
+        def git_branch
+          exec_git_command('git rev-parse --abbrev-ref HEAD')
+        rescue => e
+          Datadog.logger.debug("Unable to read git branch: #{e.message} at #{e.backtrace.first}")
+          nil
+        end
+
+        def git_commit_sha
+          exec_git_command('git rev-parse HEAD')
+        rescue => e
+          Datadog.logger.debug("Unable to read git commit SHA: #{e.message} at #{e.backtrace.first}")
+          nil
+        end
+
+        def git_tag
+          exec_git_command('git tag --points-at HEAD')
+        rescue => e
+          Datadog.logger.debug("Unable to read git tag: #{e.message} at #{e.backtrace.first}")
+          nil
+        end
+
+        def git_base_directory
+          exec_git_command('git rev-parse --show-toplevel')
+        rescue => e
+          Datadog.logger.debug("Unable to read git base directory: #{e.message} at #{e.backtrace.first}")
+          nil
+        end
+
+        def exec_git_command(cmd)
+          out, status = Open3.capture2e(cmd)
+
+          raise out unless status.success?
+
+          return nil if out.empty?
+
+          out.strip # There's always a "\n" at the end of the command output
+        end
+
+        def extract_local_git
+          env = {
+            TAG_WORKSPACE_PATH => git_base_directory,
+            Datadog::Ext::Git::TAG_REPOSITORY_URL => git_repository_url,
+            Datadog::Ext::Git::TAG_COMMIT_SHA => git_commit_sha,
+            Datadog::Ext::Git::TAG_BRANCH => git_branch,
+            Datadog::Ext::Git::TAG_TAG => git_tag,
+            Datadog::Ext::Git::TAG_COMMIT_MESSAGE => git_commit_message
+          }
+
+          if (commit_users = git_commit_users)
+            env.merge!(
+              Datadog::Ext::Git::TAG_COMMIT_AUTHOR_NAME => commit_users[:author_name],
+              Datadog::Ext::Git::TAG_COMMIT_AUTHOR_EMAIL => commit_users[:author_email],
+              Datadog::Ext::Git::TAG_COMMIT_AUTHOR_DATE => commit_users[:author_date],
+              Datadog::Ext::Git::TAG_COMMIT_COMMITTER_NAME => commit_users[:committer_name],
+              Datadog::Ext::Git::TAG_COMMIT_COMMITTER_EMAIL => commit_users[:committer_email],
+              Datadog::Ext::Git::TAG_COMMIT_COMMITTER_DATE => commit_users[:committer_date]
```

Comment:
Minor: Consider using `.fetch(...)` since we expect all of these to be available.

Reply:
I'll follow up on this one on the refactor task that is coming next.